### DISCUSSION TITLE
Are there plans to support 32 bit ARM architecture (e.g. older Raspberry Pis, several SBCs)?

### DISCUSSION THREAD

**Discussion by blakek** [pkgxdev/pkgx#456] [Help]
👋🏻 Hi, as the title says, I was curious if 32 bit ARM support could be added.

Currently, when trying to install (e.g. using `sh <(curl https://tea.xyz)`), one will get this output:

```
tea: error: (currently) unsupported OS or architecture (Linux/armv6l)
let’s talk about it: https://github.com/orgs/teaxyz/discussions
```

From reading the [install script](https://github.com/teaxyz/setup/blob/main/install.sh), it seems like `gum` and the tea CLI would need a new compile target added (e.g. `armel` or `armhf`). If that's what's needed, I'd be happy to help (with a little pointing in right direction). However, I realize I'm missing all the packaging details and there may be a lot more needed.

Thanks!

**Comment by mxcl** [comment]
We'd love to support it. It's just a matter of how we can do the builds.

The install script is not required to use tea, you can just download the binary if we have built it for your platform.

/cc @jhheider

**Reply by jhheider** [reply]
And you should be able to build tea for any environment deno supports (but some tweaks will be needed to the CLI sources).

The hard part for full binary support will be build hardware. GitHub only has x86-64; azure or AWS may have arm32 vms we could set up the runner on (if GitHub provides actions-runner for arm32). Otherwise, it'd be a lot of cross-compiling, which will almost certainly go wrong for stuff other than deno/rust/etc.

Worth investigation, certainly.
